extensions/resource-deployment/src/services/tools/azCliTool.ts

```diff
@@ -40,18 +42,98 @@ export class AzCliTool extends ToolBase {
 		return true;
 	}
 
-	public install(): Promise<void> {
-		return Promise.reject('not implemented');
+	protected async getInstallationPath(): Promise<string | undefined> {
+		switch (this.osType) {
+			case OsType.linux:
+				return defaultInstallationRoot;
+			default:
+				return win32InstallationRoot;
+		}
 	}
 
+	readonly allInstallationCommands: Map<OsType, Command[]> = new Map<OsType, Command[]>([
+		[OsType.linux, linuxInstallationCommands],
+		[OsType.win32, win32InstallationCommands],
+		[OsType.darwin, macOsInstallationCommands],
+		[OsType.others, defaultInstallationCommands]
+	]);
+
 	protected getVersionFromOutput(output: string): SemVer | undefined {
 		if (output && output.includes('azure-cli')) {
 			return new SemVer(output.split(EOL)[0].replace('azure-cli', '').replace(/ /g, '').replace('*', ''));
 		} else {
 			return undefined;
 		}
 	}
-	protected get versionCommand(): string {
-		return 'az --version';
+	protected get versionCommand(): Command {
+		return {
+			command: 'az --version'
+		};
 	}
 }
+
+const win32InstallationCommands = [
+	{
+		comment: localize('resourceDeployment.AziCli.DeletingPreviousAzureCli.msi', 'deleting previously downloaded azurecli.msi if one exists ...'),
+		command: `IF EXIST .\\AzureCLI.msi DEL /F .\\AzureCLI.msi`
+	},
+	{
+		sudo: true,
+		comment: localize('resourceDeployment.AziCli.DownloadingAndInstallingAzureCli', 'downloading azurecli.msi and installing azure-cli ...'),
+		command: `powershell -Command "& {(New-Object System.Net.WebClient).DownloadFile('https://aka.ms/installazurecliwindows', 'AzureCLI.msi'); Start-Process msiexec.exe -Wait -ArgumentList '/I AzureCLI.msi /passive /quiet /lvx ADS_AzureCliInstall.log'}"`
+	},
+	{
+		comment: localize('resourceDeployment.AziCli.DisplayingInstallationLog', 'displaying the installation log ...'),
+		command: `type AzureCliInstall.log | findstr /i /v /c:"cached product context" | findstr /i /v /c:"has no eligible binary patches" `,
+		ignoreError: true
+	}
+];
+const macOsInstallationCommands = [
+	{
+		comment: localize('resourceDeployment.AziCli.UpdatingBrewRepository', 'updating your brew repository for azure-cli installation ...'),
```

Comment:
all these places are using single quote for actual string, please update to double quotes.

Reply:
will do.